src/store/sagas/positions.ts

```diff
@@ -311,6 +294,177 @@ export function* handleInitPositionWithSOL(action: PayloadAction<InitPositionDat
   }
 }
 
+function* handleInitPositionWithSOL(action: PayloadAction<InitPositionData>): Generator {
+  const data = action.payload
+
+  if (
+    (data.tokenX.toString() === WRAPPED_SOL_ADDRESS && data.xAmount === 0) ||
+    (data.tokenY.toString() === WRAPPED_SOL_ADDRESS && data.yAmount === 0)
+  ) {
+    return yield* call(handleInitPosition, action)
+  }
+
+  // To initialize both the pool and position, separate transactions are necessary, as a single transaction does not have enough memory to accommodate all instructions for both pool and position creation with SOL.
```

Comment:
comment will be clearer:
does not have enough **memory** -> does not have enough **space**